src/test/java/org/sagebionetworks/bridge/exporter/integration/ExportTest.java

```diff
@@ -0,0 +1,1142 @@
+        // case of endDateTime is equal to the last export date time
+        assertExport(exporterRequest, null, uploadId, 0, false, false, true);
+
+        // then test case of endDateTime before lastExportDateTime -- there should be no change as well
+        exporterRequest = JSON_OBJECT_MAPPER.createObjectNode();
+        exporterRequest.put("exportType", ExportType.HOURLY.name());
+        exporterRequest.put("endDateTime", dateTimeBeforeExport.minusHours(1).toString());
+        studyWhitelistArray = JSON_OBJECT_MAPPER.createArrayNode();
+        studyWhitelistArray.add(studyId);
+        exporterRequest.set("studyWhitelist", studyWhitelistArray);
+
+        sqsHelper.sendMessageAsJson(exporterSqsUrl, JSON_OBJECT_MAPPER.writeValueAsString(exporterRequest), 0);
+
+        Item lastExportDateTime = ddbExportTimeTable.getItem("studyId", studyId);
+        Long lastExportDateTimeEpoch = lastExportDateTime.getLong("lastExportDateTime");
+        assertNotNull(lastExportDateTimeEpoch);
+
+        assertEquals((long)lastExportDateTimeEpoch, dateTimeBeforeExport.getMillis());
+    }
+
+    @Test
+    public void testS3Override() throws Exception {
+        UploadValidationStatus
+                uploadStatus = user.getClient(ForConsentedUsersApi.class).getUploadStatus(uploadId).execute().body();
+
+        // first upload a test override file into s3 bucket
+        s3FileName = "integ-test-redrive-record-ids." + DateTime.now().withZone(DateTimeZone.UTC).toString();
+        List<String> recordIds = ImmutableList.of(uploadStatus.getRecord().getId());
+        s3Helper.writeLinesToS3(recordIdOverrideBucket, s3FileName, recordIds);
+
+        DateTime dateTimeBeforeExport = DateTime.now();
+        LOG.info("Time before request daily exporting: " + dateTimeBeforeExport.toString());
+
+        // then upload another test file
+        createAndEncryptTestFiles(TEST_FILES_A_2, TEST_FILES_B_2);
+        String uploadId2 = testUpload("./legacy-survey-encrypted");
+
+        ObjectNode exporterRequest = JSON_OBJECT_MAPPER.createObjectNode();
+        exporterRequest.put("endDateTime", dateTimeBeforeExport.toString());
+        ArrayNode studyWhitelistArray = JSON_OBJECT_MAPPER.createArrayNode();
+        studyWhitelistArray.add(studyId);
+        exporterRequest.set("studyWhitelist", studyWhitelistArray);
+        exporterRequest.put("recordIdS3Override", s3FileName);
+
+        // assert only the first upload was exported
+        assertExport(exporterRequest, null, uploadId, 0, false, true, false);
+    }
+
+    private void assertExport(ObjectNode exporterRequestOri, ExportType exportType, String uploadId,
+            int offset, boolean exportAll, boolean ignoreLastExportDateTime, boolean noUpload) throws IOException, InterruptedException, SynapseException {
+        // prevent accidentally export all in prod env
+        if (exportAll && isProduction(env)) return;
```

Comment:
As a safety check, we should also make sure exporterRequestOriginal has a studyWhitelist if we're in production.